devtools/test_dashboard/server.js

```diff
@@ -1,56 +1,161 @@
 var fs = require('fs');
-var http = require('http');
 var path = require('path');
-
-var browserify = require('browserify');
+var http = require('http');
 var ecstatic = require('ecstatic');
-var _open = require('open');
+var open = require('open');
+var browserify = require('browserify');
+var watchify = require('watchify');
 
-var makeWatchifiedBundle = require('../../tasks/util/make_watchified_bundle');
-var shortcutPaths = require('../../tasks/util/shortcut_paths');
 var constants = require('../../tasks/util/constants');
+var compress = require('../../tasks/util/compress_attributes');
 
+var PORT = process.argv[2] || 3000;
 
-// TODO make this an optional argument
-var PORT = '8080';
 
-var testFile;
-switch(process.argv[2]) {
-    case 'cartesian':
-        testFile = 'test_cartesian';
-        break;
-    case 'geo':
-        testFile = 'test_geo';
-        break;
-    case 'gl3d':
-        testFile = 'test_gl3d';
-        break;
-    default:
-        testFile = 'test_gl2d';
-}
-
-console.log('Using ' + testFile);
-console.log('Listening on :' + PORT + '\n');
+// Create server
+var server = http.createServer(ecstatic({
+    root: constants.pathToRoot,
+    cache: 0,
+    gzip: true
+}));
 
-// watch plotly.js
-var watchifiedBundle = makeWatchifiedBundle(function onFirstBundleCallback() {
-    _open('http://localhost:' + PORT + '/devtools/test_dashboard');
+// Bundle development source code
+var b = browserify(constants.pathToPlotlyIndex, {
+    debug: true,
+    standalone: 'Plotly',
+    transform: [compress],
+    cache: {},
+    packageCache: {},
+    plugin: [watchify]
 });
-watchifiedBundle();
-
-// build the test examples
-fs.unlink(constants.pathToTestDashboardBundle, function() {
-    browserify(path.join(__dirname, testFile), {
-        debug: true,
-        transform: [shortcutPaths]
-    })
-    .bundle(function(err) {
-        if(err) throw err;
-    })
-    .pipe(fs.createWriteStream(constants.pathToTestDashboardBundle));
+b.on('update', bundleDevtools);
+
+// Bundle devtools code
+var devtoolsPath = path.join(constants.pathToRoot, 'devtools/test_dashboard');
+var devtools = browserify(path.join(devtoolsPath, 'devtools.js'), {
+    transform: [compress]
```

Comment:
Ahh I thought that was a js-minify step or something. Serves me right for not checking.